goodput/src/worker.c

```diff
@@ -435,12 +435,22 @@ int direct_io_obwin_map(struct worker *info)
 	info->ob_valid = 1;
 
 	info->ob_ptr = mmap(NULL, info->ob_byte_cnt, PROT_READ | PROT_WRITE,
-		MAP_SHARED, info->mp_h, info->mp_h);
+		MAP_SHARED, info->mp_h, info->ob_handle);
```

Comment:
I believe ob_handle is the intended offset here.

Here and elsewhere, make sure mmap is checked vs MAP_FAILED instead of NULL.